### PR TITLE
Add cert config for custom logging target

### PR DESCRIPTION
**Pull request by aiwantaozi**
Problem:
support ssl config for custom config logging target

Solution:
add ssl fields to custom logging target

Types:
https://github.com/rancher/types/pull/702

Issue:
https://github.com/rancher/rancher/issues/17696

### CODE REVIEW

File: pkg/api/customization/logging/action.go [hunk 1]
```diff
@@ -215,14 +221,35 @@ func (h *Handler) dryRunLoggingTarget(apiContext *types.APIContext, level, clust
 	if err != nil {
 		return errors.Wrap(err, "create executor failed")
 	}
-	var stdout, stderr, stdin bytes.Buffer
+	mkdirCmd := bytes.NewBufferString("mkdir -p " + sslDir)
```
Comment (orangedeng) [inline pkg/api/customization/logging/action.go]: You need to considerate multiple dryRun actions run at the same time. You need to create a tmp dir for each action at least.

File: vendor.conf [hunk 2]
```diff
@@ -39,8 +39,9 @@ github.com/mcuadros/go-version                6d5863ca60fa6fe914b5fd43ed8533d756
 github.com/robfig/cron                        v1.1
 
 github.com/rancher/rdns-server                bf662911db6acce4d6a85d2878653f68413b9176
-github.com/rancher/norman                     196df5ed9da30dac3aa715879f44616238b8085f
-github.com/rancher/kontainer-engine           447b7766d6aa1f15986c182eb2d7b7e97ef3678e
+github.com/rancher/types                      9c5577f95855fcb7332829acf80e486b6b382934 https://github.com/aiwantaozi/types
+github.com/rancher/norman                     0557aa4ff31a3a0f007dcb1b684894f23cda390c
```
Comment (orangedeng) [inline vendor.conf]: Why are you updating norman and kontainer-engine? Does it meas you need to rebase?

File: pkg/api/customization/logging/action.go (hunk 1 above)
Comment (aiwantaozi) [reply pkg/api/customization/logging/action.go]: The fluentd command is ```fluentd -q --dry-run -i ```  ```-i``` means inline, we don't create the file, inline config just appended to the config file on-fly.

Comment (aiwantaozi) [reply pkg/api/customization/logging/action.go]: add generate temp dir for cert every time

File: vendor.conf (hunk 2 above)
Comment (aiwantaozi) [reply vendor.conf]: rebased

**Comment by aiwantaozi** [comment]
rebased

File: pkg/api/customization/logging/action.go [hunk 3]
```diff
@@ -215,14 +226,47 @@ func (h *Handler) dryRunLoggingTarget(apiContext *types.APIContext, level, clust
...
+	}
+	if clientCert != "" {
+		echoClientCertCmd := bytes.NewBufferString(`echo "` + clientCert + `" >> ` + clientCertPath)
+		errgrp.Go(func() error { return remoteExcuteCommand(executor, *echoClientCertCmd) })
+	}
+	if clientKey != "" {
+		echoClientKeyCmd := bytes.NewBufferString(`echo "` + clientKey + `" >> ` + clientKeyPath)
+		errgrp.Go(func() error { return remoteExcuteCommand(executor, *echoClientKeyCmd) })
+	}
+	if err = errgrp.Wait(); err != nil {
```
Comment (orangedeng) [inline pkg/api/customization/logging/action.go]: Would it be better to make less api call? Maybe we can combine all the create command into one call and the test and clean-up into the other one? Or even all those command with in one api call?

Comment (aiwantaozi) [reply pkg/api/customization/logging/action.go]: updated

**Comment by orangedeng** [comment]
LGTM

File: pkg/dialer/factory.go [hunk 4]
```diff
@@ -112,7 +112,7 @@ func (f *Factory) clusterDialer(clusterName, address string) (dialer.Dialer, err
 
 	if cluster.Spec.Internal {
 		// For local (embedded, or import) we just assume we can connect directly
-		return native()
```
Comment (cjellick) [inline pkg/dialer/factory.go]: yikes! this is REALLY bad to do

Comment (cjellick) [reply pkg/dialer/factory.go]: this would have broke things in a very weird way if it got merged

File: michelia [hunk 5]
```diff
@@ -0,0 +1,2 @@
+xiaoxiao
```
Comment (cjellick) [inline michelia]: what is this file?

**Comment by aiwantaozi** [comment]
@cjellick I rebase and remove todo code careless commit in last force push, thanks

**Comment by cjellick** [comment]
LGTM

**Pull request merged**